infrastructure/code/template/details.html

```diff
@@ -542,22 +605,101 @@ <h2 class="example-fixes-header">External example fixes</h2>
 
   <!-- Scripts -->
   <script nonce="{{.Nonce}}">
-    const tabLinks = document.querySelectorAll('.tab-item');
-    const tabContents = document.querySelectorAll('.tab-content');
-    document.querySelector('.tabs-nav').addEventListener('click', (event) => {
-      const clickedTab = event.target.closest('.tab-item');
-      if (!clickedTab) return;
-
-      tabLinks.forEach(tab => {
-        tab.classList.toggle('is-selected', tab === clickedTab);
+    const ideType = 'snyk-vscode';
+
+    const MAIN_TAB_NAV_SELECTOR = '.main-tabs-nav';
+    const MAIN_TAB_ITEM_SELECTOR = '.main-tabs-nav .tab-item';
+    const MAIN_TAB_CONTENT_SELECTOR = '.tab-container > .tab-content';
+    const EXAMPLE_TAB_NAV_SELECTOR = '.example-fixes-tabs-nav';
+    const EXAMPLE_TAB_ITEM_SELECTOR = '.example-fixes-tabs-nav .tab-item';
+    const EXAMPLE_TAB_CONTENT_SELECTOR = '.example-fixes .tab-content';
+    const IGNORE_WARNING_SELECTOR = '.ignore-warning-wrapper';
+    const FIRST_EXAMPLE_CONTENT_ID = '#tab-content-0';
+    const ISSUE_OVERVIEW_MD_HEADER = '.issue-overview .overview-text h2';
+
+    document.addEventListener('DOMContentLoaded', () => {
+      const overviewHeader = document.querySelector(ISSUE_OVERVIEW_MD_HEADER);
+      if (overviewHeader) {
```

Comment:
Question: Why do we need to add this class conditionally?

Reply:
It not a conditional addition. We are making sure that we are calling `.classList.add` only if the element exists in the `DOM`. 

When we query an element that is not yet set in the `DOM`, it returns `null`, which can cause an error that might stop the execution of subsequent JavaScript. In this example, the rest of the elements are no longer responsive:

<img width="1749" alt="image" src="https://github.com/snyk/snyk-ls/assets/1948377/bd6ddf3d-3309-41a4-8da2-15f82e5ed58c">

The `DOMContentLoaded` event ensures that our script runs after the HTML is fully parsed. However, this doesn't always guarantee that every element we intend to manipulate is present. This is particularly critical for content coming from **markdown** in `additionalData.Text`, which we render in the **Overview** panel. For this case, _defensive programming_ is necessary.

Reply:
By the way, my initial intention in adding the class name `issue-overview-header` to the `<h2>` tags sourced from the markdown in `additionalData.Text` was to ensure that these headings are styled consistently with the rest of the template. 

|`additionalData.Text`|Markdown Rendered Without Additional Styles|
|-|-|
|![image](https://github.com/snyk/snyk-ls/assets/1948377/fbc49c1d-c148-483c-97bd-2ef2f1ffa70c)|<img src="https://github.com/snyk/snyk-ls/assets/1948377/d3e1218e-b96f-43c5-8716-57e8adac352c">|


However, I just realised that after your changes in #526, these headings and body will inherit the correct styles, won't they? 🙌 



|suggestion.css|
|-|
|<img width=460 src="https://github.com/snyk/snyk-ls/assets/1948377/c8a97076-c940-41dd-bd5e-d235c46dd3a8">|

Reply:
Possibly! I'll try to see if I can cleanup any redundant classes/styling